mission/mission_core/missions/ScanTheCode_mission.py

```diff
@@ -0,0 +1,219 @@
+        '''
+        Camera data is a dictionary with the following format:
+        {
+            "port": CameraData,
+            "starboard": CameraData,
+            "center": CameraData
+        }
+        CameraData is a class that contains the image frame and the results of a Yolo model (if running).
+        And can be accessed like so:
+            camera_data.frame
+            camera_data.results
+
+        PositionData is a class that contains the current position and heading of the vehicle.
+        And can be accessed like so:
+            position_data.lat : float
+            position_data.lon : float
+            position_data.position : tuple (lat, lon)
+            position_data.heading : float
+
+        The output perc_cmd has the following format (does not have to be filled):
+        {
+            "start": ["camera_name"],
+            "stop": ["camera_name"],
+            "record": ["camera_name"],
+            "stop_record": ["camera_name"],
+            "load_model": [("camera_name", "model_path")],
+            "stop_model": ["camera_name"]
+        }
+
+        The output gnc_cmd has the following format (does not have to be filled):
+        {
+            "heading": float, # The target heading for the boat to turn towards
+            "vector": tuple, # The target vector for the boat to head towards (will override heading)
+            "poshold": bool, # Whether to set the GNC to poshold mode (heading is not maintained but position is)
+            "waypoint": tuple, # The target waypoint to set the GNC to (sending any other commands will override this)
+            "end_mission": bool # Whether to end the mission
+        }
+
+        '''
+        
+        # Run one yolo inference on the provided frame at camera_data.frame
+        center_camera_data = camera_data.get("center")
+        if center_camera_data is not None:
+            # Something is detected
+            self.storageArray.append(center_camera_data.results[0].names)
+            
+            if(len(self.storageArray) != 0):
+                self.log(f"Interim processing{self.storageArray}")
+                returnVal = filter(self.storageArray)
+                records = self.analyze(returnVal)
+
+                self.colors, self.pattern = self.results(records)
+                self.log(f"Results: {self.colors[0]}, {self.colors[1]}, {self.colors[2]}")
+
+                
+                max_vote = records.get(self.pattern)
+                if max_vote is None:
+                    max_vote = 0
+                if(max_vote > 0):
+                    if(max_vote >= self.countThreshold):
+                        gnc_cmd = {"end_mission": True}
+                    else:
+                        self.light_pattern = ''.join(self.colors)
+                
+                else:
+                    # No Colors to update, can just say black
+                    self.light_pattern = ["N", "N", "N"]
```

Comment:
From what I can tell you dont actually use self.light_pattern anywhere besides for the output heartbeat. So it doesn't need to be a list. Just set it to "NNN" here.